### PR TITLE
[Actor] d_a_spotbox 99.2%

### PR DESCRIPTION
**Pull request by MrPolymorph**
image

### CODE REVIEW

File: src/d/actor/d_a_spotbox.cpp [hunk 1]
```diff
@@ -3,31 +3,78 @@
...
+    float baseScale = this->getType() != 0 ? 100.0f : 1000.0f;
+    this->mScale.x *= baseScale;
+    this->mScale.y *= baseScale;
+    this->mScale.z *= (baseScale * 1.2f);
+    this->current.pos.y += this->mScale.y * 0.5f;
+    this->mCullMtx = ((daSpotbox_c *)this)->mtx;
+    fopAcM_setCullSizeBox(this, -0.5, -0.5f, -0.5f, 0.5f, 0.5f, 0.5f);
+
+    return cPhs_COMPLEATE_e;
+}
```
Comment (MrPolymorph) [inline src/d/actor/d_a_spotbox.cpp:34]: this is the function I cant fully match due to an annoying 'mr' in the wrong place:

image

File: include/d/d_drawlist.h [hunk 2]
```diff
@@ -171,6 +171,11 @@ struct dDlst_alphaModel_c {
     void create(int);
     void set(unsigned char, float(*)[4], unsigned char);
     void draw(float(*)[4]);
+public:
+    short pad;
+    short pad2;
+    short pad3;
+    short mCount;
```
Comment (TakaRikka) [inline include/d/d_drawlist.h:178]: offset comments to left of members are preferred, and if members are temporarily unknown maybe just call them `field_0x` + offset num.
so something like
```cpp
/* 0x0 */ s16 field_0x0;
/* 0x2 */ s16 field_0x2;
/* 0x4 */ s16 field_0x4;
/* 0x6 */ s16 mCount;
```

File: src/d/actor/d_a_spotbox.cpp [hunk 3]
```diff
@@ -3,31 +3,78 @@
...
+int daSpotbox_c::create() {
+    /* Matching 96% */
+    fopAcM_SetupActor(this, daSpotbox_c);
+    float baseScale = this->getType() != 0 ? 100.0f : 1000.0f;
+    this->mScale.x *= baseScale;
+    this->mScale.y *= baseScale;
+    this->mScale.z *= (baseScale * 1.2f);
+    this->current.pos.y += this->mScale.y * 0.5f;
+    this->mCullMtx = ((daSpotbox_c *)this)->mtx;
+    fopAcM_setCullSizeBox(this, -0.5, -0.5f, -0.5f, 0.5f, 0.5f, 0.5f);
```
Comment (TakaRikka) [inline src/d/actor/d_a_spotbox.cpp:31]: `this->` is not necessary in c++ if you're working in a class function

File: src/d/actor/d_a_spotbox.cpp [hunk 4]
```diff
@@ -3,31 +3,78 @@
...
+        g_dComIfG_gameInfo.drawlist.setAlphaModel2(0x3, this->mtx, 0x20);
+    }
+    return TRUE;
+}
+
+int daSpotbox_c::execute() {
+    PSMTXTrans(mDoMtx_stack_c::now, this->current.pos.x, this->current.pos.y, this->current.pos.z);
+    mDoMtx_YrotM(mDoMtx_stack_c::now, this->current.angle.y);
+    mDoMtx_stack_c::scaleM(this->mScale.x, this->mScale.y, this->mScale.z);
+    PSMTXCopy(mDoMtx_stack_c::now, this->mtx);
```
Comment (TakaRikka) [inline src/d/actor/d_a_spotbox.cpp:47]: these other Mtx funcs can probably be replaces by mDoMtx_stack_c inlines

**Pull request merged**